cli/src/cmd/forge/selectors.rs

```diff
@@ -89,6 +113,63 @@ impl SelectorsSubcommands {
                     }
                 }
             }
+            SelectorsSubcommands::Collision { mut first_contract, mut second_contract, build } => {
+                // Build first project
+                let first_project = build.project()?;
+                let first_outcome = if let Some(ref mut contract_path) = first_contract.path {
+                    let target_path = canonicalize(&*contract_path)?;
+                    *contract_path = target_path.to_string_lossy().to_string();
+                    compile::compile_files(&first_project, vec![target_path], true)
+                } else {
+                    compile::suppress_compile(&first_project)
+                }?;
+
+                // Build second project
+                let second_project = build.project()?;
+                let second_outcome = if let Some(ref mut contract_path) = second_contract.path {
+                    let target_path = canonicalize(&*contract_path)?;
+                    *contract_path = target_path.to_string_lossy().to_string();
+                    compile::compile_files(&second_project, vec![target_path], true)
+                } else {
+                    compile::suppress_compile(&second_project)
+                }?;
+
+                // Find the artifacts
+                let first_found_artifact = first_outcome.find_contract(&first_contract);
+                let second_found_artifact = second_outcome.find_contract(&second_contract);
+
+                // Unwrap inner artifacts
+                let first_artifact = first_found_artifact.ok_or_else(|| {
+                    eyre::eyre!("Failed to extract first artifact bytecode as a string")
+                })?;
+                let second_artifact = second_found_artifact.ok_or_else(|| {
+                    eyre::eyre!("Failed to extract second artifact bytecode as a string")
+                })?;
+
+                // Check method selectors for collisions
+                let first_method_map = first_artifact.method_identifiers.as_ref().unwrap();
+                let second_method_map = second_artifact.method_identifiers.as_ref().unwrap();
+
+                let mut colliding_methods = Vec::new();
+                for (k1, v1) in first_method_map {
```

Comment:
this is ok, but maybe we could do this in a more rusty/functional way? something like filtering whichever method map for items contained in the second and collecting this into another vector.

Reply:
sounds good! will rewrite this with a functional construct 👍